Error: config] [ERROR] Failures: 
Error: config] [ERROR]   ConfigChangeAspectTest.testRemoveConfigByIdWithRpcType:438 expected: <PUBLISH_BY_RPC> but was: <REMOVE_BY_HTTP>
[nacos-config] [INFO] 
Error: config] [ERROR] Tests run: 1569, Failures: 1, Errors: 0, Skipped: 0

这个单测没有过，好像是你修改的部分，麻烦再看一下

--- a/config/src/main/java/com/alibaba/nacos/config/server/aspect/ConfigChangeAspect.java
+++ b/config/src/main/java/com/alibaba/nacos/config/server/aspect/ConfigChangeAspect.java
@@ -24,6 +24,9 @@ import com.alibaba.nacos.config.server.model.gray.BetaGrayRule;
 import com.alibaba.nacos.config.server.model.gray.TagGrayRule;
 import com.alibaba.nacos.config.server.utils.ConfigExecutor;
 import com.alibaba.nacos.config.server.utils.TimeUtils;
+import com.alibaba.nacos.core.context.RequestContext;
+import com.alibaba.nacos.core.context.RequestContextHolder;
+import com.alibaba.nacos.core.context.addition.BasicContext;
 import com.alibaba.nacos.plugin.config.ConfigChangePluginManager;
 import com.alibaba.nacos.plugin.config.constants.ConfigChangeConstants;
 import com.alibaba.nacos.plugin.config.constants.ConfigChangeExecuteTypes;
@@ -44,6 +47,7 @@ import java.util.Locale;
 import java.util.Properties;
 
 import static com.alibaba.nacos.config.server.constant.Constants.HTTP;
+import static com.alibaba.nacos.config.server.constant.Constants.RPC;
 
 /**
  * Config change pointcut aspect,which config change plugin services will pointcut.
@@ -112,14 +116,8 @@ public class ConfigChangeAspect {
             grayRuleExp = tag;
         }
         
-        ConfigChangePointCutTypes configChangePointCutType = null;
-        if (HTTP.equals(scrType)) {
-            // via console or api calls
-            configChangePointCutType = ConfigChangePointCutTypes.PUBLISH_BY_HTTP;
-        } else {
-            // via sdk rpc calls
-            configChangePointCutType = ConfigChangePointCutTypes.PUBLISH_BY_RPC;
-        }
+        ConfigChangePointCutTypes configChangePointCutType =
+            resolvePublishPointCutType(scrType);
         final List<ConfigChangePluginService> pluginServices = getPluginServices(
             configChangePointCutType);
         // didn't enabled or add relative plugin
@@ -158,14 +156,8 @@ public class ConfigChangeAspect {
         final String srcUser = (String) args[5];
         final String scrType = (String) args[6];
         
-        ConfigChangePointCutTypes configChangePointCutType = null;
-        if (HTTP.equals(scrType)) {
-            // via console or api calls
-            configChangePointCutType = ConfigChangePointCutTypes.PUBLISH_BY_HTTP;
-        } else {
-            // via sdk rpc calls
-            configChangePointCutType = ConfigChangePointCutTypes.PUBLISH_BY_RPC;
-        }
+        ConfigChangePointCutTypes configChangePointCutType =
+            resolveRemovePointCutType(scrType);
         final List<ConfigChangePluginService> pluginServices =
             getPluginServices(configChangePointCutType);
         // didn't enabled or add relative plugin
@@ -273,6 +265,64 @@ public class ConfigChangeAspect {
             }
         }
         return new ArrayList<>();
+    }
+    
+    private ConfigChangePointCutTypes resolvePublishPointCutType(String srcType) {
+        String resolvedSourceType = resolveSourceType(srcType);
+        if (HTTP.equals(resolvedSourceType)) {
+            // via console or api calls
+            return ConfigChangePointCutTypes.PUBLISH_BY_HTTP;
+        }
+        if (RPC.equals(resolvedSourceType)) {
+            // via sdk rpc calls
+            return ConfigChangePointCutTypes.PUBLISH_BY_RPC;
+        }
+        logUnknownSourceType(srcType);
+        return ConfigChangePointCutTypes.PUBLISH_BY_UNKNOWN;
+    }
+    
+    private ConfigChangePointCutTypes resolveRemovePointCutType(String srcType) {
+        String resolvedSourceType = resolveSourceType(srcType);
+        if (HTTP.equals(resolvedSourceType)) {
+            // via console or api calls
+            return ConfigChangePointCutTypes.REMOVE_BY_HTTP;
+        }
+        if (RPC.equals(resolvedSourceType)) {
+            // via sdk rpc calls
+            return ConfigChangePointCutTypes.REMOVE_BY_RPC;
+        }
+        logUnknownSourceType(srcType);
+        return ConfigChangePointCutTypes.REMOVE_BY_UNKNOWN;
+    }
+    
+    private String resolveSourceType(String srcType) {
+        RequestContext requestContext = RequestContextHolder.getContext();
+        if (requestContext != null) {
+            String requestProtocol = requestContext.getBasicContext().getRequestProtocol();
+            if (BasicContext.HTTP_PROTOCOL.equals(requestProtocol)) {
+                return HTTP;
+            }
+            if (BasicContext.GRPC_PROTOCOL.equals(requestProtocol)) {
+                return RPC;
+            }
+        }
+        if (HTTP.equals(srcType) || RPC.equals(srcType)) {
+            return srcType;
+        }
+        return null;
+    }
+    
+    private void logUnknownSourceType(String srcType) {
+        RequestContext requestContext = RequestContextHolder.getContext();
+        String requestProtocol = null;
+        String requestTarget = null;
+        if (requestContext != null) {
+            requestProtocol = requestContext.getBasicContext().getRequestProtocol();
+            requestTarget = requestContext.getBasicContext().getRequestTarget();
+        }
+        LOGGER.warn(
+            "Use unknown config change pointcut due to unknown source type: {}, requestProtocol: {}, requestTarget: {}",
+            srcType, requestProtocol, requestTarget);
     }
     
     private boolean isEnabled(ConfigChangePluginService configChangePluginService) {

--- a/config/src/main/java/com/alibaba/nacos/config/server/controller/v3/ConfigControllerV3.java
+++ b/config/src/main/java/com/alibaba/nacos/config/server/controller/v3/ConfigControllerV3.java
@@ -229,6 +229,7 @@ public class ConfigControllerV3 {
         
         ConfigRequestInfo configRequestInfo = new ConfigRequestInfo();
         configRequestInfo.setSrcIp(RequestUtil.getRemoteIp(request));
+        configRequestInfo.setSrcType(Constants.HTTP);
         configRequestInfo.setRequestIpApp(RequestUtil.getAppName(request));
         configRequestInfo.setBetaIps(request.getHeader("betaIps"));
         configRequestInfo.setCasMd5(request.getHeader("casMd5"));

--- a/config/src/test/java/com/alibaba/nacos/config/server/aspect/ConfigChangeAspectTest.java
+++ b/config/src/test/java/com/alibaba/nacos/config/server/aspect/ConfigChangeAspectTest.java
@@ -22,6 +22,8 @@ import com.alibaba.nacos.config.server.configuration.ConfigChangeConfigs;
 import com.alibaba.nacos.config.server.model.ConfigRequestInfo;
 import com.alibaba.nacos.config.server.model.form.ConfigForm;
 import com.alibaba.nacos.config.server.utils.RequestUtil;
+import com.alibaba.nacos.core.context.RequestContextHolder;
+import com.alibaba.nacos.core.context.addition.BasicContext;
 import com.alibaba.nacos.plugin.config.ConfigChangePluginManager;
 import com.alibaba.nacos.plugin.config.constants.ConfigChangeConstants;
 import com.alibaba.nacos.plugin.config.constants.ConfigChangeExecuteTypes;
@@ -80,6 +82,9 @@ class ConfigChangeAspectTest {
     
     @BeforeEach
     void before() {
+        RequestContextHolder.getContext().getBasicContext().setRequestProtocol(null);
+        RequestContextHolder.getContext().getBasicContext().setRequestTarget(null);
+        
         //mock config change service enabled.
         propertiesStatic = Mockito.mockStatic(PropertiesUtil.class);
         requestUtilMockedStatic = Mockito.mockStatic(RequestUtil.class);
@@ -105,6 +110,9 @@ class ConfigChangeAspectTest {
     
     @AfterEach
     void after() {
+        RequestContextHolder.getContext().getBasicContext().setRequestProtocol(null);
+        RequestContextHolder.getContext().getBasicContext().setRequestTarget(null);
+        
         propertiesStatic.close();
         requestUtilMockedStatic.close();
         ConfigChangePluginManager.reset();
@@ -269,6 +277,114 @@ class ConfigChangeAspectTest {
     }
     
     @Test
+    void testRequestProtocolTakesPrecedenceOverSourceTypeForPublish() throws Throwable {
+        RequestContextHolder.getContext().getBasicContext()
+            .setRequestProtocol(BasicContext.HTTP_PROTOCOL);
+        RequestContextHolder.getContext().getBasicContext().setRequestTarget("POST /v3/cs/config");
+        Mockito.when(configChangePluginService.executeType())
+            .thenReturn(ConfigChangeExecuteTypes.EXECUTE_BEFORE_TYPE);
+        when(pjp.getArgs()).thenReturn(new Object[] {configForm, configRequestInfo});
+        when(configRequestInfo.getSrcType()).thenReturn("rpc");
+        when(configForm.getDataId()).thenReturn("dataId");
+        when(pjp.proceed(any())).thenReturn("Success");
+        
+        configChangeAspect.publishOrUpdateConfigAround(pjp);
+        
+        ArgumentCaptor<ConfigChangeRequest> requestCaptor =
+            ArgumentCaptor.forClass(ConfigChangeRequest.class);
+        verify(configChangePluginService).execute(requestCaptor.capture(), any());
+        assertEquals(ConfigChangePointCutTypes.PUBLISH_BY_HTTP,
+            requestCaptor.getValue().getRequestType());
+    }
+    
+    @Test
+    void testRequestProtocolTakesPrecedenceOverSourceTypeForRemove() throws Throwable {
+        RequestContextHolder.getContext().getBasicContext()
+            .setRequestProtocol(BasicContext.GRPC_PROTOCOL);
+        RequestContextHolder.getContext().getBasicContext().setRequestTarget("ConfigRemoveRequest");
+        Mockito.when(configChangePluginService.executeType())
+            .thenReturn(ConfigChangeExecuteTypes.EXECUTE_BEFORE_TYPE);
+        when(pjp.getArgs()).thenReturn(
+            new Object[] {"dataId", "group", "namespaceId", null, "127.0.0.1", "nacos", "http"});
+        when(pjp.proceed(any())).thenReturn("Success");
+        
+        configChangeAspect.removeConfigByIdAround(pjp);
+        
+        ArgumentCaptor<ConfigChangeRequest> requestCaptor =
+            ArgumentCaptor.forClass(ConfigChangeRequest.class);
+        verify(configChangePluginService).execute(requestCaptor.capture(), any());
+        assertEquals(ConfigChangePointCutTypes.REMOVE_BY_RPC,
+            requestCaptor.getValue().getRequestType());
+    }
+    
+    @Test
+    void testPublishWithUnknownSourceTypeUsesUnknownPointcut() throws Throwable {
+        Mockito.when(configChangePluginService.executeType())
+            .thenReturn(ConfigChangeExecuteTypes.EXECUTE_BEFORE_TYPE);
+        when(pjp.getArgs()).thenReturn(new Object[] {configForm, configRequestInfo});
+        when(configRequestInfo.getSrcType()).thenReturn(null);
+        when(configForm.getDataId()).thenReturn("dataId");
+        when(pjp.proceed(any())).thenReturn("Success");
+        
+        configChangeAspect.publishOrUpdateConfigAround(pjp);
+        
+        ArgumentCaptor<ConfigChangeRequest> requestCaptor =
+            ArgumentCaptor.forClass(ConfigChangeRequest.class);
+        verify(configChangePluginService).execute(requestCaptor.capture(), any());
+        assertEquals(ConfigChangePointCutTypes.PUBLISH_BY_UNKNOWN,
+            requestCaptor.getValue().getRequestType());
+    }
+    
+    @Test
+    void testRemoveWithUnknownSourceTypeUsesUnknownPointcut() throws Throwable {
+        Mockito.when(configChangePluginService.executeType())
+            .thenReturn(ConfigChangeExecuteTypes.EXECUTE_BEFORE_TYPE);
+        when(pjp.getArgs()).thenReturn(
+            new Object[] {"dataId", "group", "namespaceId", null, "127.0.0.1", "nacos", null});
+        when(pjp.proceed(any())).thenReturn("Success");
+        
+        configChangeAspect.removeConfigByIdAround(pjp);
+        
+        ArgumentCaptor<ConfigChangeRequest> requestCaptor =
+            ArgumentCaptor.forClass(ConfigChangeRequest.class);
+        verify(configChangePluginService).execute(requestCaptor.capture(), any());
+        assertEquals(ConfigChangePointCutTypes.REMOVE_BY_UNKNOWN,
+            requestCaptor.getValue().getRequestType());
+    }
+    
+    @Test
+    void testRemoveHttpSourceTypeUsesRemovePointcut() throws Throwable {
+        Mockito.when(configChangePluginService.executeType())
+            .thenReturn(ConfigChangeExecuteTypes.EXECUTE_BEFORE_TYPE);
+        when(pjp.getArgs()).thenReturn(
+            new Object[] {"dataId", "group", "namespaceId", null, "127.0.0.1", "nacos", "http"});
+        when(pjp.proceed(any())).thenReturn("Success");
+        
+        configChangeAspect.removeConfigByIdAround(pjp);
+        ArgumentCaptor<ConfigChangeRequest> requestCaptor =
+            ArgumentCaptor.forClass(ConfigChangeRequest.class);
+        verify(configChangePluginService).execute(requestCaptor.capture(), any());
+        assertEquals(ConfigChangePointCutTypes.REMOVE_BY_HTTP,
+            requestCaptor.getValue().getRequestType());
+    }
+    
+    @Test
+    void testRemoveRpcSourceTypeUsesRemovePointcut() throws Throwable {
+        Mockito.when(configChangePluginService.executeType())
+            .thenReturn(ConfigChangeExecuteTypes.EXECUTE_BEFORE_TYPE);
+        when(pjp.getArgs()).thenReturn(
+            new Object[] {"dataId", "group", "namespaceId", null, "127.0.0.1", "nacos", "rpc"});
+        when(pjp.proceed(any())).thenReturn("Success");
+        
+        configChangeAspect.removeConfigByIdAround(pjp);
+        ArgumentCaptor<ConfigChangeRequest> requestCaptor =
+            ArgumentCaptor.forClass(ConfigChangeRequest.class);
+        verify(configChangePluginService).execute(requestCaptor.capture(), any());
+        assertEquals(ConfigChangePointCutTypes.REMOVE_BY_RPC,
+            requestCaptor.getValue().getRequestType());
+    }
+    
+    @Test
     void testPublishConfigWithBetaIps() throws Throwable {
         Mockito.when(configChangePluginService.executeType())
             .thenReturn(ConfigChangeExecuteTypes.EXECUTE_AFTER_TYPE);
@@ -325,7 +441,7 @@ class ConfigChangeAspectTest {
         ArgumentCaptor<ConfigChangeRequest> captor =
             ArgumentCaptor.forClass(ConfigChangeRequest.class);
         verify(configChangePluginService).execute(captor.capture(), any());
-        assertEquals(ConfigChangePointCutTypes.PUBLISH_BY_RPC,
+        assertEquals(ConfigChangePointCutTypes.REMOVE_BY_RPC,
             captor.getValue().getRequestType());
     }
     

--- a/console/src/main/java/com/alibaba/nacos/console/controller/v3/config/ConsoleConfigController.java
+++ b/console/src/main/java/com/alibaba/nacos/console/controller/v3/config/ConsoleConfigController.java
@@ -134,6 +134,7 @@ public class ConsoleConfigController {
         
         ConfigRequestInfo configRequestInfo = new ConfigRequestInfo();
         configRequestInfo.setSrcIp(RequestUtil.getRemoteIp(request));
+        configRequestInfo.setSrcType(Constants.HTTP);
         configRequestInfo.setRequestIpApp(RequestUtil.getAppName(request));
         configRequestInfo.setBetaIps(request.getHeader("betaIps"));
         configRequestInfo.setCasMd5(request.getHeader("casMd5"));

--- a/plugin/config/src/main/java/com/alibaba/nacos/plugin/config/constants/ConfigChangePointCutTypes.java
+++ b/plugin/config/src/main/java/com/alibaba/nacos/plugin/config/constants/ConfigChangePointCutTypes.java
@@ -32,6 +32,10 @@ public enum ConfigChangePointCutTypes {
      */
     PUBLISH_BY_RPC("publishOrUpdateByRpc"),
     /**
+     * Publish config with unknown source type.
+     */
+    PUBLISH_BY_UNKNOWN("publishOrUpdateByUnknown"),
+    /**
      * Remove by id through http.
      */
     REMOVE_BY_HTTP("removeSingleByHttp"),
@@ -39,6 +43,10 @@ public enum ConfigChangePointCutTypes {
      * Remove through rpc.
      */
     REMOVE_BY_RPC("removeSingleByRpc"),
+    /**
+     * Remove with unknown source type.
+     */
+    REMOVE_BY_UNKNOWN("removeSingleByUnknown"),
     /**
      * Import config file through http/console.
      */

